### PR TITLE
Azure Storage Key Detector plugin

### PR DESCRIPTION
**Pull request by DariuszPorowski**
This plugin searches for Azure Storage Account access keys.

Co-authored-by: Dariusz Porowski <Dariusz.Porowski@microsoft.com>

### CODE REVIEW

**Review by domanchi — APPROVED** [review]

File: tests/plugins/azure_storage_key_test.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        [
+            (
+                'AccountKey=lJzRc1YdHaAA2KCNJJ1tkYwF/+mKK6Ygw0NGe170Xu592euJv2wYUtBlV8z+qnlcNQSnIYVTkLWntUO1F8j8rQ==',
+                True,
+            ),
+        ],
+    )
+    def test_analyze(self, payload, should_flag):
+        logic = AzureStorageKeyDetector()
+        output = logic.analyze_line(payload, 1, 'mock_filename')
```
Comment (domanchi) [inline tests/plugins/azure_storage_key_test.py:19]: Do you mind merging this into our `pre-v1-launch` branch instead? Unless you have a specific desire to have this on master sooner, we're current undergoing a large effort to change up the underlying architecture of this tool, and it just helps reduce merge conflicts by making changes to that branch instead.

All you would need to do is 

```
assert logic.analyze_line(filename='mock_filename', line=payload)
```

as such https://github.com/Yelp/detect-secrets/blob/pre-v1-launch/tests/plugins/slack_test.py, then point this PR to the that branch instead.

Comment (DariuszPorowski) [reply tests/plugins/azure_storage_key_test.py:19]: @domanchi OK, thank for feedback. I am going to work on it.

Comment (DariuszPorowski) [reply tests/plugins/azure_storage_key_test.py]: @domanchi updated.

**Review by domanchi — APPROVED** [review]

**Comment by domanchi** [comment]
Looks like pre-commit failed. Be sure to run `pre-commit run` to fix up any linting issues you may have, so that the codebase can be of consistent style.

**Comment by DariuszPorowski** [comment]
@domanchi Make sense. Done :)

**Comment by domanchi** [comment]
Tests pass -- just looks like py35 fails (as expected). Merging.

**Comment by DariuszPorowski** [comment]
Hey @domanchi Quick question. Do you have any timeline for v1 release?

**Comment by domanchi** [comment]
Honestly, this is up to one other project maintainer to review the changes. I think @KevinHock is looking at it now, but we're all kinda working on this on the side. I'd like to say sooner rather than later, as there are many features in v1 that would be fantastic to be deployed (and several new features I have in the pipeline for it), but I don't have a set date.

**Pull request merged**